src/rgw/rgw_crypt.cc

```diff
@@ -902,7 +916,13 @@ static int get_actual_key_from_kms(CephContext *cct,
 
   map<string, string>::iterator it = str_map.find(std::string(key_id));
   if (it != str_map.end() ) {
-    std::string master_key = from_base64((*it).second);
+    boost::string_view master_key_sview((*it).second);
+    if (!is_valid_base64_encoded(master_key_sview)) {
+      ldout(cct, 5) << "ERROR: get_actual_key_from_kms invalid encryption key id \
+                        which contains charactor that is not base64 encoded." << dendl;
+      return -EINVAL;
+    }
+    std::string master_key = from_base64(master_key_sview);
```

Comment:
`from_base64()` will throw exceptions on bad input, so i think we're just missing the exception handling here. can you try this instead?
```c++
    std::string master_key;
    try {
      master_key = from_base64(it->second);
    } catch (...) {
      ldout(cct, 5) << "ERROR: get_actual_key_from_kms invalid encryption key id \
                        which contains charactor that is not base64 encoded." << dendl;
      return -EINVAL;
    }
```

Reply:
Yes, you are right. I have recommit and added exception handling for from_base64() on bad input. Do you mind to review it again? Thank you~